openhcl/virt_mshv_vtl/src/processor/tdx/mod.rs

```diff
@@ -1543,9 +1547,70 @@ impl UhProcessor<'_, TdxBacked> {
             }
         }
 
+        // First, check that the VM entry was even successful.
+        let vmx_exit = exit_info.code().vmx_exit();
+        if vmx_exit.vm_enter_failed() {
+            match vmx_exit.basic_reason() {
+                VmxExitBasic::BAD_GUEST_STATE => {
```

Comment:
Can you move this case into a function, to get it out of the reviewer hot path?

Reply:
you mean all the logging, or something else?

Reply:
I am hoping for something like.

```
if vmx_exit.vm_enter_failed() {
    return Err(HypervisorError(self.handle_vm_enter_failed(vmx_exit)));
}
```

Reply:
sure i can move stuff around to its own fn.